### PR TITLE
Improvements to chat

### PR DESCRIPTION
**Pull request by bonham000**
**Fixes various things in chat and also tries to setup `MongoStore` for session persistence on server.**

Frontend routing doesn't work in some cases...
- Trying to navigate to a user profile from within chat was not working and I couldn't figure out how to do it.
- Typing some URLs in directly does not work.

### CODE REVIEW

**Comment by no-stack-dub-sack** [comment]
> Trying to navigate to a user profile from within chat was not working and I couldn't figure out how to do it. 

@bonham000 do you mean the new routes like `dashboard/profile/bonham000`?

If so, what did you try to make that work so that I do not duplicate your efforts when I get home?

**Comment by bonham000** [comment]
@no-stack-dub-sack what I tried:

- Pass callback to `Chat` component and call on any click of a user avatar: In callback, `this.props.history.push(/url...)` or `this.props.history.replace(/url...)` did not work, `url` being any combination of the target `url` (`/dashboard/profile/:user`). I could get the actual `url` to update correctly but the page would not navigate.
- Try rendering the Avatar image in a `Link` or `NavLink` component that directs to the page. Same issue, no actual routing.

**Note:** maybe this has to do with how you configured the page to load the data — i.e. we would need to dispatch an action to update the Redux store for this view. In this case:

**My two cents:** we have all users in `community` piece of state. Why not navigate to `profile/:user` and in `mapStateToProps` just extract the username from the routing `props` and then get this user from the `community` in state. No need to incorporate a new reducer and new piece of state for this and any additional actions. Just navigate to a page and we show 1 user based on your `url`. Sorry now I've said this in both of my notes here... but again look at the `mapStateToProps` in `ChatController`.

**Comment by no-stack-dub-sack** [comment]
@bonham000 So, this - `const { username } = props.match.params;` - pulls `username` corrects because it is matching `:username` from here - 
`<Route exact path={`${url}/chat/:username`} component={Chat}/>`?

Or is that not right? I haven't logged the code or read the docs yet - just venturing from looking at how this is set up...

File: client/src/reducers/privateChat.js [hunk 1]
```diff
@@ -67,7 +67,7 @@ export default (state = Map(), action) => {
       return state.updateIn([reciepient, 'history'], l => {
```
Comment (no-stack-dub-sack) [inline client/src/reducers/privateChat.js]: @bonham000 may be a little late for this for this PR, seeing as I just merged it, and it only matters if other people will eventually be looking at this code base, but I noticed a little typo here: `reciepient` should actually be `recipient`.

**Pull request merged**